chain/bitcoind_conn.go

```diff
@@ -472,6 +539,45 @@ func (c *BitcoindConn) RemoveClient(id uint64) {
 	delete(c.rescanClients, id)
 }
 
+// isBlockPrunedErr determines if the error returned by the GetBlock RPC
+// corresponds to the requested block being pruned.
+func isBlockPrunedErr(err error) bool {
+	rpcErr, ok := err.(*btcjson.RPCError)
+	return ok && rpcErr.Code == btcjson.ErrRPCMisc &&
+		rpcErr.Message == errBlockPrunedStr
+}
+
+// GetBlock returns a raw block from the server given its hash. If the server
+// has already pruned the block, it will be retrieved from one of its peers.
+func (c *BitcoindConn) GetBlock(hash *chainhash.Hash) (*wire.MsgBlock, error) {
+	block, err := c.client.GetBlock(hash)
+	if !isBlockPrunedErr(err) || c.prunedBlockDispatcher == nil {
```

Comment:
I think this would be easier to read if we just immediatly return the block if `err == nil`, then go on to handle the error cases.

Reply:
Done.